### DISCUSSION TITLE
prisma migrations in production

### DISCUSSION THREAD

**Discussion by torohaifisch** [prisma/prisma#17213] [Q&A] [answered]
hello everyone, i wanted to ask about the prisma migration commands.
i read the documentation about it and i know it is recommended to run  prisma migrate deploy in production inside a ci pipeline.
but i have a question, considering starting with an empty database, is there any conflict  if i run prisma migrate dev to populate everything and then prisma migrate deploy? or like reset and then deploy?

**Comment by pantharshit00** [comment, marked as answer]
Hello @torohaifisch,

If your database is empty, it is fine to run `npx prisma migrate dev`. The difference between `dev` and `deploy` is that, `dev` will prompt you to create a new migration if it finds changes in the schema and `deploy` will not prompt for any migration creation, it will just try to apply any pending migrations in the migrations folder. 

Also, `dev` will **prompt** you to accept any data loss if it detects a drift so that it can start a new migrations history, but `deploy` will just simply abort and ask you to resolve conflicts.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
